### ISSUE TITLE
Unable to install atom-netconf 3.3.0 package on atom 1.58.0 on ubuntu 18.04 VM

### ISSUE THREAD

**Issue by selvaitech** [nokia/atom-netconf#14]
Unable to install atom-netconf 3.3.0 package on atom version 1.58.0

Error Shown in atom GUI:

Installing atom-netconf@3.3.0 failed.
undefined

No package is installed.

**Comment by wisotzky** [comment]
I've just fix a typo in one of my files.
Could you please try it again?

New version is 3.3.1

**Comment by wisotzky** [comment]
Fixed by:
https://github.com/nokia/atom-netconf/commit/53bd8c3fec2cd3dd0a01c3480cd6a8d157a012dc#diff-76b5a357391276b282a516f54f48ef3c207f46d8192dc58c208d5183d38415f8

**Comment by selvaitech** [comment]
same issue with 3.3.1

Installing atom-netconf@3.3.1 failed.Hide output
undefined

no package is installed again

**Comment by wisotzky** [comment]
@selvaitech, could you please share the stack trace?

**Comment by selvaitech** [comment]
@wisotzky , please find the stack trace attached.

UI shows atom-netconf package version as 3.3.1 but when installed stack trace shows "package 3.2.0". can you please check?

[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.58.0 x64
**Electron**: 9.4.4
**OS**: Ubuntu 18.04.2
**Thrown From**: [atom-netconf](https://github.com/nokia/atom-netconf) package 3.2.0


### Stack Trace

Failed to load the atom-netconf package

```
At request.charAt is not a function

TypeError: request.charAt is not a function
    at Function.Module._resolveLookupPaths (internal/modules/cjs/loader.js:632:15)
    at Module._resolveFilename (internal/modules/cjs/loader.js:780:20)
    at o._resolveFilename (electron/js2c/renderer_init.js:43:689)
    at Function.get_Module._resolveFilename (/usr/share/atom/resources/app/static/<embedded>:11:157920)
    at Module.require (/app.asar/static/index.js:61:43)
    at require (/usr/share/atom/resources/app/static/<embedded>:11:149207)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/toolbar-netconf.coffee:13:27)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at Module.get_Module._compile (/usr/share/atom/resources/app/static/<embedded>:11:149891)
    at Object.value [as .coffee] (/usr/share/atom/resources/app/static/<embedded>:11:153485)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (/app.asar/static/index.js:72:46)
    at require (/usr/share/atom/resources/app/static/<embedded>:11:149207)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/netconf.coffee:15:18)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/netconf.coffee:1:1)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/netconf.coffee:1:1)
    at Module.get_Module._compile (/usr/share/atom/resources/app/static/<embedded>:11:149891)
    at Object.value [as .coffee] (/usr/share/atom/resources/app/static/<embedded>:11:153485)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (/app.asar/static/index.js:72:46)
    at require (internal/modules/cjs/helpers.js:74:18)
    at customRequire (/usr/share/atom/resources/app/static/<embedded>:1:756277)
    at requireModule (/usr/share/atom/resources/app/static/<embedded>:14:1195620)
    at Package.requireMainModule (/usr/share/atom/resources/app/static/<embedded>:11:3400284)
    at /usr/share/atom/resources/app/static/<embedded>:11:3386824
    at Package.measure (/usr/share/atom/resources/app/static/<embedded>:11:3385624)
    at Package.load (/usr/share/atom/resources/app/static/<embedded>:11:3386397)
    at PackageManager.loadAvailablePackage (/usr/share/atom/resources/app/static/<embedded>:11:384186)
    at PackageManager.loadPackage (/usr/share/atom/resources/app/static/<embedded>:11:382914)
    at PackageManager.activatePackage (/usr/share/atom/resources/app/static/<embedded>:11:385486)
    at /usr/share/atom/resources/app/static/<embedded>:11:915033
    at r (/usr/share/atom/resources/app/static/<embedded>:11:909639)
    at n (/usr/share/atom/resources/app/static/<embedded>:14:1194943)
    at /usr/share/atom/resources/app/static/<embedded>:14:1195097
    at Socket.<anonymous> (/usr/share/atom/resources/app/static/<embedded>:14:1193699)
    at Socket.emit (events.js:228:7)
    at Pipe.<anonymous> (net.js:664:12)
```

### Commands

```
```

### Non-Core Packages

```
atom-netconf 3.2.0 
```

**Comment by wisotzky** [comment]
@selvaitech, the issue above was raised by version "3.2.0" - which should be the one I've fixed this morning.
Could you please retry the latest version "3.3.1" and send the stack trace?

**Comment by selvaitech** [comment]
> @selvaitech, the issue above was raised by version "3.2.0" - which should be the one I've fixed this morning.
> Could you please retry the latest version "3.3.1" and send the stack trace?

I have uninstalled atom 1.58.0 and re-installed. Tried to load the package 3.3.1 and above error message is shown.

**Comment by wisotzky** [comment]
That's strange... The error clearly indicates that it tried to load 3.2.0 (but not 3.3.1). So would be curious why it's not loading the right version. What's the procedure you are following to load 3.3.1?

**Comment by selvaitech** [comment]
> That's strange... The error clearly indicates that it tried to load 3.2.0 (but not 3.3.1). So would be curious why it's not loading the right version. What's the procedure you are following to load 3.3.1?

![image](https://user-images.githubusercontent.com/19651638/127856888-d57c48d5-6404-4fbe-ae9a-0f22672c6b3c.png)

Here is what i get when i tried to check the package.

**Comment by wisotzky** [comment]
Installation of packages is independent of installation of ATOM.
Could you try the following:

- Execute `rm -rf ~/.atom/packages/atom-netconf`
- Restart ATOM
- Install `atom-netconf` again

**Comment by selvaitech** [comment]
> Installation of packages is independent of installation of ATOM.
> Could you try the following:
> 
> * Execute `rm -rf ~/.atom/packages/atom-netconf`
> * Restart ATOM
> * Install `atom-netconf` again

followed the steps mentioned above, failing to install the package with below stack trace.

TypeError: request.charAt is not a function
    at Function.Module._resolveLookupPaths (internal/modules/cjs/loader.js:632:15)
    at Module._resolveFilename (internal/modules/cjs/loader.js:780:20)
    at o._resolveFilename (electron/js2c/renderer_init.js:43:689)
    at Function.get_Module._resolveFilename (/usr/share/atom/resources/app/static/<embedded>:11:157920)
    at Module.require (file:///usr/share/atom/resources/app.asar/static/index.js:61:43)
    at require (/usr/share/atom/resources/app/static/<embedded>:11:149207)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/toolbar-netconf.coffee:13:27)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at Module.get_Module._compile (/usr/share/atom/resources/app/static/<embedded>:11:149891)
    at Object.value [as .coffee] (/usr/share/atom/resources/app/static/<embedded>:11:153485)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (file:///usr/share/atom/resources/app.asar/static/index.js:72:46)
    at require (/usr/share/atom/resources/app/static/<embedded>:11:149207)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/netconf.coffee:15:18)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/netconf.coffee:1:1)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/netconf.coffee:1:1)
    at Module.get_Module._compile (/usr/share/atom/resources/app/static/<embedded>:11:149891)
    at Object.value [as .coffee] (/usr/share/atom/resources/app/static/<embedded>:11:153485)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (file:///usr/share/atom/resources/app.asar/static/index.js:72:46)
    at require (internal/modules/cjs/helpers.js:74:18)
    at customRequire (/usr/share/atom/resources/app/static/<embedded>:1:756277)
    at requireModule (/usr/share/atom/resources/app/static/<embedded>:14:1195620)
    at Package.requireMainModule (/usr/share/atom/resources/app/static/<embedded>:11:3400284)
    at /usr/share/atom/resources/app/static/<embedded>:11:3386824

Before installing package version shows as below,
![image](https://user-images.githubusercontent.com/19651638/127858814-e95b5f65-42b9-45be-abd2-2620a0df97bc.png)

Post that error, below is the status:

![image](https://user-images.githubusercontent.com/19651638/127857723-1dde1b2f-1e7a-40b2-a693-a9dc5244e93d.png)

**Comment by selvaitech** [comment]
@wisotzky , any updates on this issue?
On windows 10 laptop I  am able to install 1.58.0 & atom-netconf : 3.3.1 ---> I am able to connect to my netconf server and can send netconf requests. 

On Ubuntu 18.04

Uninstalled atom using sudo apt remove atom
sudo rm -rf .atom/
Freshly installed atom 1.58.0 version(sudo apt install atom)
Tried installing atom-netconf-3.3.1 and got below eception

TypeError: request.charAt is not a function
    at Function.Module._resolveLookupPaths (internal/modules/cjs/loader.js:632:15)
    at Module._resolveFilename (internal/modules/cjs/loader.js:780:20)
    at o._resolveFilename (electron/js2c/renderer_init.js:43:689)
    at Function.get_Module._resolveFilename (/usr/share/atom/resources/app/static/<embedded>:11:157920)
    at Module.require (file:///usr/share/atom/resources/app.asar/static/index.js:61:43)
    at require (/usr/share/atom/resources/app/static/<embedded>:11:149207)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/toolbar-netconf.coffee:13:27)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at Module.get_Module._compile (/usr/share/atom/resources/app/static/<embedded>:11:149891)
    at Object.value [as .coffee] (/usr/share/atom/resources/app/static/<embedded>:11:153485)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (file:///usr/share/atom/resources/app.asar/static/index.js:72:46)
    at require (/usr/share/atom/resources/app/static/<embedded>:11:149207)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/netconf.coffee:15:18)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/netconf.coffee:1:1)
    at Object.<anonymous> (/home/srama006/atom-netconf/lib/netconf.coffee:1:1)
    at Module.get_Module._compile (/usr/share/atom/resources/app/static/<embedded>:11:149891)
    at Object.value [as .coffee] (/usr/share/atom/resources/app/static/<embedded>:11:153485)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (file:///usr/share/atom/resources/app.asar/static/index.js:72:46)
    at require (internal/modules/cjs/helpers.js:74:18)
    at customRequire (/usr/share/atom/resources/app/static/<embedded>:1:756277)
    at requireModule (/usr/share/atom/resources/app/static/<embedded>:14:1195620)
    at Package.requireMainModule (/usr/share/atom/resources/app/static/<embedded>:11:3400284)
    at Package.activateNow (/usr/share/atom/resources/app/static/<embedded>:11:3388268)
    at /usr/share/atom/resources/app/static/<embedded>:11:3388046
    at Package.measure (/usr/share/atom/resources/app/static/<embedded>:11:3385624)
    at /usr/share/atom/resources/app/static/<embedded>:11:3387906
    at new Promise (<anonymous>)
    at Package.activate (/usr/share/atom/resources/app/static/<embedded>:11:3387849)
    at PackageManager.activatePackage (/usr/share/atom/resources/app/static/<embedded>:11:385612)
    at /usr/share/atom/resources/app/static/<embedded>:11:915033
    at r (/usr/share/atom/resources/app/static/<embedded>:11:909639)
    at n (/usr/share/atom/resources/app/static/<embedded>:14:1194943)
    at /usr/share/atom/resources/app/static/<embedded>:14:1195097
    at Socket.<anonymous> (/usr/share/atom/resources/app/static/<embedded>:14:1193699)
    at Socket.emit (events.js:228:7)
    at Pipe.<anonymous> (net.js:664:12)

After the error, when I checked installed package I see package status as below.

![image](https://user-images.githubusercontent.com/19651638/128461943-d0518a9c-2f21-4c15-9ce1-a01080df4e8a.png)

**Comment by wisotzky** [comment]
@selvaitech, interesting... The screenshot clearly says that atom-netconf 3.2.0 was installed but not 3.3.1... The actual error shown in the logs was fixed in 3.3.1 but not 3.2.0. So need to figure out, what makes your system loading 3.2.0 but not 3.3.1. Once this is fixed, your issues should be solved.

**Comment by wisotzky** [comment]
Hi @selvaitech, please check if the new Visual Studio Code extension can be used.
Check README for details.
